src/v/datalake/schema_registry.cc

```diff
@@ -22,23 +26,31 @@ get_schema_id_result get_value_schema_id(iobuf& buf) {
     if (buf.size_bytes() < sizeof(uint8_t) + sizeof(int32_t)) {
         return get_schema_error::not_enough_bytes;
     }
-    iobuf_const_parser parser(buf);
-    auto magic = parser.consume_type<uint8_t>();
-    if (magic != schema_registry_magic_byte) {
-        return get_schema_error::no_schema_id;
-    }
-    auto id = parser.consume_be_type<int32_t>();
-    schema_message_data res = {
-      .schema_id = pandaproxy::schema_registry::schema_id{id},
-      .shared_message_data = buf.share(
-        parser.bytes_consumed(), parser.bytes_left())};
+    try {
+        iobuf_const_parser parser(buf);
+        auto magic = parser.consume_type<uint8_t>();
+        if (magic != schema_registry_magic_byte) {
+            return get_schema_error::no_schema_id;
+        }
+        auto id = parser.consume_be_type<int32_t>();
+        schema_message_data res = {
+          .schema_id = pandaproxy::schema_registry::schema_id{id},
+          .shared_message_data = buf.share(
+            parser.bytes_consumed(), parser.bytes_left())};
 
-    return res;
+        return res;
+    } catch (...) {
```

Comment:
Here and elsewhere: Are we assuming `std::out_of_range` is the only possible exception? If so, you might consider catching that one explicitly. 

If the catchall is needed for `noexcept` confidence, maybe a different `get_schema_error` would make sense?

Reply:
Chatted offline, gonna punt on this given we have a warn log for visibility